### PR TITLE
[BEAM-2613] add integration test for comparison operators

### PR DESCRIPTION
**Pull request by xumingming**
Summary:

1. added integration test for comparison operators.
2. refactored comparison operator expressions into a separate package: `comparison`.

### CODE REVIEW

**Comment by xumingming** [comment]
R: @XuMingmin @jbonofre  @takidau

File: dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlComparisonOperatorsIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * Integration test for comparison operators.
+ */
+public class BeamSqlComparisonOperatorsIntegrationTest
+    extends BeamSqlBuiltinFunctionsIntegrationTestBase {
+  @Test
+  public void testComparisionOperators() throws Exception {
+    ExpressionChecker checker = new ExpressionChecker()
+        .addExpr("1 = 1", true)
```
Comment (XuMingmin) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlComparisonOperatorsIntegrationTest.java]: can you use data fields in the expressions? That helps to ensure the methods work good for different types.

Comment (xumingming) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlComparisonOperatorsIntegrationTest.java]: You mean to test all the data types for all the comparison operators?

Comment (XuMingmin) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlComparisonOperatorsIntegrationTest.java]: Right, I do find some issues in MATH functions, it's safer to do so.

**Comment by xumingming** [comment]
Updates:

1. Added more test cases.
2. Renamed `BeamSqlLargerThanExpression` to `BeamSqlGreaterThanExpression`.

**Comment by xumingming** [comment]
Retest this please

**Comment by xumingming** [comment]
Retest this please

**Comment by XuMingmin** [comment]
retest this please

**Comment by xumingming** [comment]
R: @JingsongLi

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/comparison/BeamSqlGreaterThanEqualExpression.java [hunk 2]
```diff
@@ -15,16 +15,17 @@
...
+import org.apache.beam.dsls.sql.interpreter.operator.BeamSqlExpression;
 
 /**
  * {@code BeamSqlExpression} for {@code >=} operation.
  */
-public class BeamSqlLargerThanEqualExpression extends BeamSqlCompareExpression {
+public class BeamSqlGreaterThanEqualExpression extends BeamSqlCompareExpression {
 
-  public BeamSqlLargerThanEqualExpression(List<BeamSqlExpression> operands) {
+  public BeamSqlGreaterThanEqualExpression(List<BeamSqlExpression> operands) {
```
Comment (JingsongLi) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/comparison/BeamSqlGreaterThanEqualExpression.java]: BeamSqlGreaterThanOrEqualExpression maybe more suitable? Others is the same.

File: dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlComparisonOperatorsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        Arrays.asList(
+            "c_tinyint_0", "c_tinyint_1", "c_tinyint_2",
+            "c_smallint_0", "c_smallint_1", "c_smallint_2",
+            "c_integer_0", "c_integer_1", "c_integer_2",
+            "c_bigint_0", "c_bigint_1", "c_bigint_2",
+            "c_float_0", "c_float_1", "c_float_2",
+            "c_double_0", "c_double_1", "c_double_2",
+            "c_decimal_0", "c_decimal_1", "c_decimal_2",
+            "c_varchar_0", "c_varchar_1", "c_varchar_2",
+            "c_boolean_0", "c_boolean_1"
```
Comment (JingsongLi) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlComparisonOperatorsIntegrationTest.java]: c_boolean_0 to  c_boolean_false?

File: dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlComparisonOperatorsIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+        .addExpr("c_varchar_2 > c_varchar_1", true)
+        .addExpr("c_varchar_1 > c_varchar_1", false)
+        .addExpr("c_varchar_1 > c_varchar_2", false)
+        ;
+
+    checker.buildRunAndCheck();
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testGreaterThan_exception() {
```
Comment (JingsongLi) [inline dsls/sql/src/test/java/org/apache/beam/dsls/sql/integrationtest/BeamSqlComparisonOperatorsIntegrationTest.java]: testGreaterThan_exception to testGreaterThanException?

**Comment by JingsongLi** [comment]
Merged, feel free to close.

**Comment by xumingming** [comment]
Thanks @XuMingmin @JingsongLi

**Pull request closed without merging**